test/__snapshots__/TypeScriptGenerator-test.ts.snap

```diff
@@ -1684,7 +1682,7 @@ export type RelayClientIDFieldQueryResponse = {
         readonly id: string;
         readonly commentBody?: {
             readonly __id: string;
-            readonly __typename: string;
+            readonly __typename: "PlainCommentBody" | "MarkdownCommentBody";
```

Comment:
https://github.com/facebook/relay/blob/43ae013b9aacf63e19126932100bcb8bdf495603/packages/relay-test-utils-internal/testschema.graphql#L234-L248

Reply:
Shouldn't the correct snapshot be the following:

```ts
{
  commentBody?: {
    readonly __id: string;
    readonly __typename: "PlainCommentBody";
    readonly  text?: {
      readonly __id: string;
      readonly __typename: "Text";
      readonly text: string | null;
    } | null;
  } | {
    readonly __id: string;
    readonly __typename: "MarkdownCommentBody";
  } | null;
}
```

Reply:
In general I would suggest comparing the test output with the official tests. We should strive to have the same behavior as much as possible. This way when there's bugs we can always do a 1-1 comparison with the official type generation.

Reply:
@n1ru4l Yes, you're right. I think this isn't working as well as it should yet because of the check whether the typename is the only base field being selected. That part makes that code quite confusing to be honest.

@kastermester I'm not really familiar with the output for Flow as I don't use it, is it better than the current Typescript output?